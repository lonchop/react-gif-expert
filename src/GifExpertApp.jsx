import { useState } from 'react'
import { AddCategory, GifGrid } from './components'
import { v4 as uuidv4 } from 'uuid';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch'])

  const onAddCategory = (newCategory) => {
    setCategories([newCategory, ...categories])
  }

  return (
    <>
      <h1>Gif Expert App</h1>
      <AddCategory onNewCategory={onAddCategory} />
      {categories.map((category) => (
        <GifGrid
          key={uuidv4()}
          category={category}
        />
      ))}
    </>
  )
}
