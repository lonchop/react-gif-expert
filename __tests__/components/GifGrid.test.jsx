/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react'
import { GifGrid } from '../../src/components'
import { useFetchGif } from '../../src/hooks/useFetchGif'
jest.mock('../../src/hooks/useFetchGif')

describe('Pruebas en <GifGrid />', () => {
  const category = 'One Punch'
  test('Debe hacer match con el snapshot', () => {
    useFetchGif.mockReturnValue({
      images: [],
      isLoading: true,
    })

    const { container } = render(<GifGrid category={category} />)

    expect(container).toMatchSnapshot()
  })

  test('Debe mostrar el loading inicialmente', () => {
    useFetchGif.mockReturnValue({
      images: [],
      isLoading: true,
    })

    render(<GifGrid category={category} />)
    expect(screen.getByText('Cargando...'))
  })

  test('Debe mostrar items cuando cargan imÃ¡genes useFetchGif', () => {
    const gif = [
      {
        id: 1,
        title: 'Saitama',
        url: 'http://localhost/image1.jpg',
      },
      {
        id: 2,
        title: 'Genos',
        url: 'http://localhost/image2.jpg',
      },
    ]

    useFetchGif.mockReturnValue({
      images: gif,
      isLoading: false,
    })

    render(<GifGrid category={category} />)
    screen.debug()
    expect(screen.getAllByRole('img').length).toBe(2)
  })
})
