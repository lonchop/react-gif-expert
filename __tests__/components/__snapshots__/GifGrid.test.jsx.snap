// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe hacer match con el snapshot 1`] = `
<div>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
